commcare_connect/form_receiver/processor.py

```diff
@@ -195,16 +195,20 @@ def process_deliver_unit(user, xform: XForm, app: CommCareApp, opportunity: Oppo
             or datetime.date.today() > claim.end_date
         ):
             user_visit.status = VisitValidationStatus.over_limit
+            user_visit.status_modification_date = now()
```

Comment:
I would suggest to create an `update_status` method on the `UserVisit`.

```
def update_status(status):
     self.status = status
     self.status_modification_date = now()
```
Then you can replace most of the updates without having to update the date in each place.

There is also property setter way of doing this for e.g. https://stackoverflow.com/a/57523694

Reply:
I have reviewed the information provided in this link:  https://stackoverflow.com/questions/2898711/django-model-fields-getter-setter/57523694#57523694. If I use the property decorator, I will need to change the property name from status to _status in the model. This change will require updating any queries where the status property is used. Am I correct in this understanding? Should I proceed with this approach, or should I consider defining a method instead?